jena-arq/src/test/java/org/apache/jena/sparql/function/scripting/TestNV.java

```diff
@@ -58,7 +61,15 @@ public class TestNV {
         assertEquals(nv.getLabel(), nv.getValue());
         assertEquals("BlankNode", nv.getTermType());
     }
-    
+
+    @Test public void nv_14() {
+        String largeInteger = "1" + "0".repeat(22);
+        NodeValue nodeValue = NV.toNodeValue(new BigInteger(largeInteger).doubleValue());
+        NV nv = new NV(nodeValue);
+        assertEquals(nv.getValue(), largeInteger);
```

Comment:
Arguments the wrong way round? It's "expected" , "actual".

(This test does not work for your original example but that's OK because very large numbers will always be a problem between double and exact representation.)